src/course-outline/data/api.js

```diff
@@ -105,3 +108,107 @@ export async function restartIndexingOnCourse(reindexLink) {
 
   return camelCaseObject(data);
 }
+
+/**
+ * Get course section
+ * @param {string} sectionId
+ * @returns {Promise<Object>}
```

Comment:
Can you put the actual type here? `Object` is not very useful.

Nit/tip: Also, it's not your code but there are some major problems with the JSDoc comments in the other functions above. if you put `// @ts-check` on the first line of this function, you can easily see all the errors using an IDE like VS Code.

e.g.

![Screenshot 2023-12-08 at 3 00 45 PM](https://github.com/openedx/frontend-app-course-authoring/assets/945577/f63626ea-60fb-4e00-b7fb-5eb0eef5e504)

Should be:

```ts
/**
 * Get course best practices.
 * @param {{courseId: string, excludeGraded: boolean, all: boolean}} options
 * @returns {Promise<{isSelfPaced: boolean, sections: any }>}
 */
export async function getCourseBestPractices({
  courseId,
  excludeGraded,
  all,
}) {
```

You don't have to change that now, but I do strongly recommend putting in the `// @ts-check` so that you're not adding more errors.

Reply:
@bradenmacdonald Thanks! Actually, even this is not my code :sweat_smile:. But yes, I should have checked these docs. Will try to fix them, atleast in course-outline related files.

Reply:
haha I see. Well thanks for making those fixes anyways :)
I'm not planning to do any further review here, since @KristinAoki has it covered. But let me know if you do have specific questions I can help with.